packages/kit/src/runtime/server/page/csp.js

```diff
@@ -0,0 +1,197 @@
+import { escape_html_attr } from '../../../utils/escape.js';
+import { sha256, base64 } from './crypto.js';
+
+/** @type {Promise<void>} */
+export let csp_ready;
+
+/** @type {() => string} */
+let generate_nonce;
+
+/** @type {(input: string) => string} */
+let generate_hash;
+
+if (typeof crypto !== 'undefined') {
+	const array = new Uint8Array(16);
```

Comment:
Is reusing the array safe and a perf improvement? otherwise might be cleaner to have it inside the function

Reply:
From https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues:

> All elements in the array are overwritten with random numbers

So it's safe. I would expect the perf impact to be minimal but there's no real reason to keep allocating new memory and garbage collecting it